app/assets/javascripts/editor.js

```diff
@@ -85,45 +85,50 @@ function snippet_changed(item){
   $('#association').toggle(!checked);
 }
 
-function create_editor(item) {
-  item.parent().prepend("<div id='editor1'></div>");
-  item.hide();
-  Editor = ace.edit("editor1");
+function create_editor() {
+  var editorId = "editor" + Math.floor((Math.random() * 100) + 1),
```

Comment:
> What would it mean to write it in Angular? /me personally don't object
> better ways to solve problems? Would it require the entire bastion plugin
> to support it?

No, any html element can be an angular "app" as long as the angular javascript is included.  So these pages themselves could be an angular "app"s just so they could use the directive I would create for this editor.

As far as including the Angular JS source goes, we wouldn't necessarily have to include bastion in order to get this to work.  A stop-gap could be to include the angular source independently, we could even use something like [angularjs-rails](https://github.com/hiravgandhi/angularjs-rails).